src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,189 @@ void VulnerabilityScannerFacade::start(
                 std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
-                    SyscollectorDeltas::GetDelta(message.data());
+                    delta;
 
-                scanOrchestrator->run(data);
+                try
+                {
+                    scanOrchestrator->run(data);
+                }
+                catch (const OsDataException& e)
+                {
+                    m_deltaQueue->push(std::move(std::string(message.data(), message.size())));
+                    logDebug1(WM_VULNSCAN_LOGTAG,
+                              "%s. dbsync_packages events postponed: %s package. Queue size: %d",
+                              e.what(),
+                              (delta->data_as_dbsync_packages() && delta->data_as_dbsync_packages()->name())
+                                  ? delta->data_as_dbsync_packages()->name()->c_str()
+                                  : "",
+                              m_deltaQueue->size());
+                }
+                catch (const std::exception& e)
+                {
+                    logError(WM_VULNSCAN_LOGTAG, "ScanOrchestrator::run::Exception: %s", e.what());
+                }
             });
 
         // Subscription to syscollector rsync events.
         m_syscollectorRsyncSubscription =
             std::make_unique<RouterSubscriber>("rsync-syscollector", "vulnerability_scanner_rsync");
         m_syscollectorRsyncSubscription->subscribe(
-            [scanOrchestrator](const std::vector<char>& message)
+            [scanOrchestrator, this](const std::vector<char>& message)
             {
-                std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
+                const SyscollectorSynchronization::SyncMsg* syncMsg =
                     SyscollectorSynchronization::GetSyncMsg(message.data());
+                std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
+                    syncMsg;
 
-                scanOrchestrator->run(data);
+                try
+                {
+                    scanOrchestrator->run(data);
+                }
+                catch (const OsDataException& e)
+                {
+                    m_syncQueue->push(std::move(std::string(message.data(), message.size())));
+                    logDebug1(WM_VULNSCAN_LOGTAG,
+                              "%s. syscollector_packages event postponed: %s package. Queue size: %d",
+                              e.what(),
+                              (syncMsg->data_as_state() &&
+                               syncMsg->data_as_state()->attributes_as_syscollector_packages() &&
+                               syncMsg->data_as_state()->attributes_as_syscollector_packages()->name())
+                                  ? syncMsg->data_as_state()->attributes_as_syscollector_packages()->name()->c_str()
```

Comment:
This expression is really hard to follow, can we declare it as a const variable before the debug call with a descriptive name?

Reply:
We should unify criteria for these cases, I know creating unneeded variables should be avoided, that's why I did this way. Also, I was told early in this PR to not create an extra variable, but I don't know if the degradation impact is measurable in some way. 
About the const modifier, I don't see a way to use it and avoid the ternary operator I'm using now that caused the readability complexity. 
Let me know if I'm missing something.